rpcs3/Emu/Cell/lv2/lv2.cpp

```diff
@@ -1162,16 +1162,34 @@ bool lv2_obj::awake_unlocked(cpu_thread* cpu, s32 prio)
 
 			if (const auto ppu = g_ppu[i]; ppu == cpu)
 			{
-				if (g_ppu[i + 1]->prio != ppu->prio)
+				std::size_t j = i + 1;
+
+				for (; j < g_ppu.size(); j++)
 				{
+					if (g_ppu[j]->prio != ppu->prio)
+					{
+						break;
+					}
+				}
+
+				if (j == i + 1)
+				{
+					// Empty 'same prio' threads list
 					return false;
 				}
-				else
+
+				// Rotate current thread to the last position of the 'same prio' threads list
+				std::rotate(g_ppu.begin() + i, g_ppu.begin() + 1, g_ppu.begin() + j);
```

Comment:
I'm not sure rotate works here. Second arg should be within [i,j) range or I misunderstand something.

Reply:
yep its a typo, second arg shiuld be i + 1.